common/functions/src/scalars/expressions/binary.rs

```diff
@@ -32,84 +38,67 @@ where F: Fn(L::RefType<'_>, R::RefType<'_>, &mut EvalContext) -> O
     }
 }
 
-/// A common struct to caculate binary expression scalar op.
-#[derive(Clone)]
-pub struct ScalarBinaryExpression<L: Scalar, R: Scalar, O: Scalar, F> {
-    func: F,
-    _phantom: PhantomData<(L, R, O)>,
-}
-
-impl<L: Scalar, R: Scalar, O: Scalar, F> ScalarBinaryExpression<L, R, O, F>
-where F: ScalarBinaryFunction<L, R, O>
+pub fn scalar_binary_op<L: Scalar, R: Scalar, O: Scalar, F>(
+    l: &ColumnRef,
+    r: &ColumnRef,
+    f: F,
+    ctx: &mut EvalContext,
+) -> Result<<O as Scalar>::ColumnType>
+where
+    F: ScalarBinaryFunction<L, R, O>,
```

Comment:
I think we can remove `ScalarBinaryFunction`.

Just use `F: Fn(L::RefType<'a>, R::RefType<'a>, &mut EvalContext) -> O::RefType<'a>` instead